site-src/concepts/versioning.md

```diff
@@ -1,116 +1,86 @@
-    * Everything that is valid in a patch release
-    * New experimental API fields or resources
-    * Loosened validation
-    * Making required fields optional
-    * Removal of experimental fields
-    * Removal of experimental resources
-    * Graduation of fields or resources from experimental to stable track
-* MAY introduce a new **API version** with a new bundle minor version, which may
-  include:
-    * Everything that is valid in a minor release
-    * Renamed fields
-    * Anything else that is valid in a new Kubernetes API version
-    * Removal/tombstoning of beta fields
-* MAY release a new major bundle version (v1.0) as part of graduating the API to
-  GA and releasing a new API version.
-
-Note that each new bundle version, no matter how small, may include updated
-CRDs, webhook, or both. Implementations may read annotations on Gateway API CRDs
-(defined below) to determine the version and channel of CRDs that have been
-installed in the cluster.
-
-### Implementers:
-* MUST handle fields with loosened validation without crashing
-* MUST handle fields that have transitioned from required to optional without
-  crashing
-* MUST NOT rely on webhook or CRD validation as a security mechanism. If field
-  values need to be escaped to secure an implementation, both webhook and CRD
-  validation can be bypassed and cannot be relied on. Instead, implementations
-  should implement their own escaping or validation as necessary. To avoid
-  duplicating work, Gateway API maintainers are considering adding a shared
-  validation package that implementations can use for this purpose. This is
-  tracked by [#926](https://github.com/kubernetes-sigs/gateway-api/issues/926).
-
-### Installers:
-* MUST install a full Gateway API bundle, with matching CRD and webhook
-  versions.
-
-## Adding Experimental Fields
-Over time, it will be useful to add experimental fields to the API. In upstream
-Kubernetes, those would generally be guarded with feature gates. With Gateway
-API we will accomplish by releasing experimental versions of our CRDs.
-
-With this approach, we achieve a similar result. Instead of using feature gates
-and validation to prevent fields from being set, we just release separate CRDs.
-Once the API reaches beta, each bundle release can include 2 sets of CRDs,
-stable and experimental.
-
-New fields will be added to the experimental set of CRDs first, and may graduate
-to stable APIs later. Experimental fields will be marked with the
-`+experimental` annotation in Go type definitions. Gateway API CRD generation
-will exclude these fields from stable CRDs. Experimental fields may be removed
-from the API. Due to the experimental nature of these CRDs, they are not
-recommended for production use.
-
-If experimental fields are removed or renamed, the original field name should be
-removed from the go struct, with a tombstone comment ensuring the field name
-will not be reused. 
+### API Versions (e.g. v1alpha2, v1beta1)
```

Comment:
One thing I found odd from trying to put together some Gateway API bits for the sig-network annual report (kubernetes/community#6614) is that we have nice release notes for ... uh ... releases, but not for API versions, which is probably what folks should be more interested in?

Just a side observation, not something to address in this PR. Maybe something to think about for when v1beta1 is cut?

Reply:
I think that's actually by design. An API version like v1beta1 really just indicates the stability of a resource, it's fairly rare for major changes to occur at the same time an API version is graduating to a new stability level. Of course when we transitioned from v1alpha1 to v1alpha2 there were pretty significant release notes included with v0.4.0, but I think most API version bumps from here on out are going to be bumping stability, not making significant changes. Instead, each new release of Gateway API will likely include new features, and I think that's what the release notes should focus on. Does that seem reasonable?